sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java

```diff
@@ -377,6 +379,18 @@ public ConfigurationClientBuilder serviceVersion(ConfigurationServiceVersion ver
         return this;
     }
 
+    /**
+     * Sets the customized headers that is used when each request is sent. It is possible to use new header value for
+     * each request.
+     *
+     * @param headersSupplier user's headers supplier applied to each request.
+     * @return The updated ConfigurationClientBuilder object.
+     */
+    public ConfigurationClientBuilder headersSupplier(Supplier<HttpHeaders> headersSupplier) {
```

Comment:
How is this expected to be used? This is a very different API than we have seen in other builders.

Reply:
User provides supplier that we call every time for each request.

Reply:
I am not sure if this is the right approach. How will this ensure thread-safety? If the same client is making several concurrent service calls, how are the headers generated and tracked by this supplier?